src/core/zipfile.cpp

```diff
@@ -20,16 +20,40 @@
 
 #include "zipfile.h"
 
+extern "C" {
+#include <libavutil/avutil.h>
+}
+
 #include "utils.h"
 
 ZipFile::ZipFile(const char *filename, const char *mode)
 : file(filename, mode, FFMS_ERROR_PARSER, FFMS_ERROR_FILE_READ)
+, is_file(true)
 , state(Initial)
 {
 	buffer.resize(65536);
 	z = {};
 }
 
+ZipFile::ZipFile()
+: file()
+, is_file(false)
+, state(Initial)
+{
+	buffer.resize(65536);
+	z = {};
+}
+
+ZipFile::ZipFile(const uint8_t *in_buffer, const size_t size)
+: file()
+, is_file(false)
+, state(Initial)
+{
+	index_buffer.resize(size);
+	memcpy(reinterpret_cast<uint8_t *>(&index_buffer[0]), in_buffer, size);
```

Comment:
These two lines can be replaced with `, index_buffer(in_buffer, in_buffer + size)` in the initializer list, which skips zero-initializing the buffer.